include/rpm/rpmpgp.h

```diff
@@ -1113,6 +1119,95 @@ int pgpDigParamsCmp(pgpDigParams p1, pgpDigParams p2);
  */
 unsigned int pgpDigParamsAlgo(pgpDigParams digp, unsigned int algotype);
 
+/** \ingroup rpmpgp
+ * Returns the issuer or the object's Key ID.
+ *
+ * If the object is a signature, then this returns the Key ID stored in the
+ * first Issuer or Issuer Fingerprint subpacket as a hex string.
+ * (This is not authenticated.)
+ *
+ * If the object is a certificate or a subkey, then this returns the key's
+ * Key ID.
+ *
+ * The caller must not free the returned buffer.
+ *
+ * param digp		parameter container
+ * return		an array of PGP_KEYID_LEN bytes.  If the issuer is
+ * 			unknown, this returns an array with all zeros.
+ */
+const uint8_t *pgpDigParamsSignID(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the primary User ID, if any.
+ *
+ * Returns the primary User ID, if any.
+ *
+ * If the object is a signature, then this returns NULL.
+ *
+ * If the object is a certificate or a subkey, then this returns the
+ * certificate's primary User ID, if any.
+ *
+ * This interface does not provide a way for the caller to recognize
+ * any embedded NUL characters.
+ *
+ * The caller must not free the returned buffer.
+ *
+ * param digp		parameter container
+ * return		a string or NULL, if there is no primary User ID.
+ */
+const char *pgpDigParamsUserID(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the object's version.
+ *
+ * Returns the object's version.
+ *
+ * If the object is a signature, then this returns the version of the
+ * signature packet.
+ *
+ * If the object is a certificate, then this returns the version of the
+ * primary key packet.
+ *
+ * If the object is a subkey, then this returns the version of the subkey's
+ * key packet.
+ *
+ * param digp		parameter container
+ * return		the object's version
+ */
+int pgpDigParamsVersion(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the object's creation time.
+ *
+ * If the object is a signature, then this returns the signature's creation
+ * time.
+ *
+ * If the object is a certificate, then this returns the primary key's key
```

Comment:
I don’t think this is what it actually does right now, but feel free to change it so that it does do this.

Reply:
That's what [the comment on the `time` field suggests](https://github.com/rpm-software-management/rpm/blob/7e7266c/rpmio/digest.h#L31), and the code [seems](https://github.com/rpm-software-management/rpm/blob/7e7266c9af883ce49b3516a5bd099d218e8e3fac/rpmio/rpmpgp.c#L483) to [agree](https://github.com/rpm-software-management/rpm/blob/7e7266c9af883ce49b3516a5bd099d218e8e3fac/rpmio/rpmpgp.c#L787).  What do you think the code does now?

Reply:
I don't think so: https://github.com/rpm-software-management/rpm/blob/7e7266c9af883ce49b3516a5bd099d218e8e3fac/rpmio/rpmpgp.c#L787

Reply:
What I meant is that when parsing public keys, `_digp->time` will wind up being overwritten by the creation time of the first signature.

Reply:
That's schizophrenic and clearly a bug.

Reply:
> That's schizophrenic and clearly a bug.

Pleas report it.

Reply:
You found the bug and you seem to understand it, so it would be better if you report it.  I've looked at the code, and I haven't figure out how that happens.

Reply:
I've removed the comment.